packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -12,12 +12,15 @@ import {
   isLogicalOrOperator,
   isNodeEqual,
   isNullLiteral,
-  isTypeFlagSet,
+  isPossiblyNullish,
   isUndefinedIdentifier,
   nullThrows,
   NullThrowsReasons,
 } from '../util';
 
+const isIdentifierOrMemberExpressionType = (type: TSESTree.AST_NODE_TYPES) =>
+  [AST_NODE_TYPES.Identifier, AST_NODE_TYPES.MemberExpression].includes(type);
```

Comment:
We have a utility we use for this (that makes a fancy type predicate)
```suggestion
const isIdentifierOrMemberExpression = isNodeOfTypes([
  AST_NODE_TYPES.Identifier,
  AST_NODE_TYPES.MemberExpression,
] as const);
```
(note that you'll need to trivially adjust the call sites)